src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -161,16 +179,28 @@
      (.remove keyboard-show-listener)
      (.remove keyboard-hide-listener))))
 
+(defn screen
+  [{:keys [initial-insets title navigation-icon recovering-keypair? render-controls]}]
+  (let [{navigation-bar-top :top} initial-insets]
+    [rn/view {:style style/full-layout}
+     [rn/keyboard-avoiding-view {:style style/page-container}
+      [quo/page-nav
+       {:margin-top navigation-bar-top
+        :background :blur
+        :icon-name  (or navigation-icon
+                        (if recovering-keypair? :i/close :i/arrow-left))
+        :on-press   #(rf/dispatch [:navigate-back])}]
+      [recovery-phrase-screen
+       {:title               title
+        :render-controls     render-controls
+        :recovering-keypair? recovering-keypair?}]]]))
+
 (defn view
   []
-  (let [{navigation-bar-top :top} (safe-area/get-insets)]
+  (let [insets (safe-area/get-insets)]
     (fn []
       (let [{:keys [recovering-keypair?]} (rf/sub [:get-screen-params])]
-        [rn/view {:style style/full-layout}
-         [rn/keyboard-avoiding-view {:style style/page-container}
-          [quo/page-nav
-           {:margin-top navigation-bar-top
-            :background :blur
-            :icon-name  (if recovering-keypair? :i/close :i/arrow-left)
-            :on-press   #(rf/dispatch [:navigate-back])}]
-          [screen recovering-keypair?]]]))))
+        [screen
+         {:title               (i18n/label :t/use-recovery-phrase)
+          :initial-insets      insets
+          :recovering-keypair? recovering-keypair?}]))))
```

Comment:
Here I've refactored a `screen` component that can be used by other screen components. Which allows us to re-use most of the logic, layout, and styles from the existing `enter-seed-phrase` screen component.